### PR TITLE
Nikhil/flagging books

### PR DESCRIPTION
**Pull request by nnarayen**

### CODE REVIEW

File: app/assets/javascripts/react_components/users/booklist_table.js.jsx [hunk 1]
```diff
@@ -56,12 +56,38 @@ var BookListTable = React.createClass({
...
+        var books = this.state.books;
+        var index = books.indexOf(book);
+        book.flagged_user_email = data.email;
+        book.flagged_user_role = data.role;
+        books.splice(index, 1, book);
+        this.setState({books: books});
+      }.bind(this),
+      error: function(xhr, status, error) {
+        toastr.error("There was an error flagging the book.");
+        console.error(this.props.url, status, err.toString());
```
Comment (varunrau) [inline app/assets/javascripts/react_components/users/booklist_table.js.jsx]: `err` is undefined here.

**Comment by varunrau** [comment]
One I problem I see with this approach is that when any action on this table hits an endpoint and replaces the books, we'll lose all the flagging state until the next refresh. So if we could move that to the backend or keep that state outside of the books, then we can avoid this problem.

**Comment by nnarayen** [comment]
Good catch on the losing flagging state on removal, check out my most recent changes. I think this solves that problem

**Comment by varunrau** [comment]
Yep that'll do it.

Could you update the flagged # in the header?

File: app/assets/stylesheets/booklists.css.scss [hunk 2]
```diff
@@ -4,6 +4,49 @@ $red-color: #d9534f;
...
+
+.flag {
+  height: 30px;
+  opacity: 0.15;
+  width: 30px;
+}
+
+.flag:hover {
+  cursor: pointer;
+  cursor: hand;
```
Comment (varunrau) [inline app/assets/stylesheets/booklists.css.scss]: yeah you should choose one here

**Comment by varunrau** [comment]
LGTM

**Comment by varunrau** [comment]
Great work! They're going to love this

**Comment by varunrau** [comment]
Also thanks for getting this done so quickly!!

**Pull request merged**